### PR TITLE
6.x sites 831 acsf d.o cjw

### PR DESCRIPTION
**Pull request by cjwest**
# NOT READY FOR REVIEW

# Summary
_See ticket:_ https://stanfordits.atlassian.net/browse/SITES-831

# Needed By Sept 14, 2018

# Criticality
- New behat tests

# Steps to Test
- Checkout this branch

# Affects 
ACSF Stack Additions

# Associated Issues and/or People
## Related JIRA ticket(s)
https://stanfordits.atlassian.net/browse/SITES-831
## Folks to notify

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by cjwest** [comment]
Hey @sherakama,
I would like you to take a look at the progress thus far and give my any feedback.
Thanks!

File: includes/features/drupal.org/addtocal/addtocal.feature [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    And I am on "admin/structure/types/manage/article/fields/field_behat_date/field-settings"
+    And I press "Save"
+    Given I am on "admin/structure/types/manage/article/display"
+    And I select "Add to Cal" from "edit-fields-field-behat-date-type"
+    And I press "Save"
+
+    # Add a date field
+    Given I am on "node/add/article"
+    And I fill in "Title" with "Behat Date Test"
+    And I press "Save"
```
Comment (sherakama) [inline includes/features/drupal.org/addtocal/addtocal.feature]: Twice?

File: includes/features/drupal.org/addtocal/addtocal.feature [hunk 2]
```diff
@@ -0,0 +1,42 @@
+Feature: Add to Cal administration module
+  In order to ensure the site is functioning as expected
+  As administrator
+  I want to verify functionality of the Add to Cal module
+
+  Background:
+    Given I am logged in as a user with the "administrator" role
+
+  @safe @api
+  Scenario: Verify module is enabled
```
Comment (sherakama) [inline includes/features/drupal.org/addtocal/addtocal.feature]: I think we can remove this check in favor of having `given module is enabled` in later scenarios.

File: includes/features/drupal.org/addtocal/addtocal.feature [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    And I am on "admin/modules"
+    Then I should see 1 "#edit-modules-other-addtocal-enable" element
+    And the "modules[Other][addtocal][enable]" checkbox should be checked
+
+  @dev @api @javascript @destructive
+  Scenario: Verify module functionality
+    Given the "addtocal" module is enabled
+    And the cache has been cleared
+    Given I am on "admin/structure/types/manage/article/fields"
```
Comment (sherakama) [inline includes/features/drupal.org/addtocal/addtocal.feature]: Tests cannot rely on the article content type as it is not on all sites. You will need to create a new test content type in the test to ensure that you can attach a field to it.

File: includes/features/drupal.org/always_visible/always_visible.feature [hunk 4]
```diff
@@ -1,10 +1,10 @@
 Feature: Always Visible
-  In order to verify functionality of the always_visble module.
-  As all types of users
-  I want to functionality
+  In order to verify functionality of the always_visible module.
+  For all types of users
+  I want to verify expected functionality
 
   @safe @api
-  Scenario: Test that the Always Visble module is enabled
```
Comment (sherakama) [inline includes/features/drupal.org/always_visible/always_visible.feature:7]: This looks like Canadian spelling :D

File: includes/features/drupal.org/special_menu_items/special_menu_items_admin.feature [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  In order to ensure the site is functioning as expected
+  As administrator
+  I want to verify functionality of the special_menu_items module
+
+  Background:
+    Given I am logged in as a user with the "administrator" role
+
+  @safe @api
+  Scenario: Verify special_menu_items module is enabled
+    Given I am on "admin/modules"
```
Comment (sherakama) [inline includes/features/drupal.org/special_menu_items/special_menu_items_admin.feature:12]: You can replace this chunk with `given the special_menu_items module is enabled`.

File: includes/features/drupal.org/special_menu_items/special_menu_items_admin.feature [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    Then I should see 1 "#edit-special-menu-items-nolink-tag" element
+
+  @safe @api
+  Scenario:  Verify that the description on the menu item edit page has changed
+    Given I am on "admin/structure/menu/manage/main-menu/add"
+    Then I should see text matching "<nolink>"
+    Then I should see text matching "<separator>"
+
+  @dev @api @javascript @destructive
+  Scenario: Verify that I can add header text that is not clickable
```
Comment (sherakama) [inline includes/features/drupal.org/special_menu_items/special_menu_items_admin.feature]: Good!

File: includes/features/drupal.org/special_menu_items/special_menu_items_admin.feature [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    Then I should be on "admin/structure/menu/manage/navigation"
+    Then I should see a ".nolink" element
+    And I should not see the link "Behat Special Menu Items <nolink>"
+    Given I am on "admin/structure/menu/manage/navigation"
+    When I click "delete" in the "Behat Special Menu Items <nolink>" row
+    And I press the "Confirm" button
+    Then I should see "The menu link Behat Special Menu Items <nolink> has been deleted."
+
+  @dev @api @javascript @destructive
+  Scenario: Verify that I can add a horizontal rule or separator
```
Comment (sherakama) [inline includes/features/drupal.org/special_menu_items/special_menu_items_admin.feature]: Good!

File: includes/features/drupal.org/addtocal/addtocal.feature (hunk 1 above)
Comment (cjwest) [reply includes/features/drupal.org/addtocal/addtocal.feature]: Yup

File: includes/features/drupal.org/always_visible/always_visible.feature (hunk 4 above)
Comment (cjwest) [reply includes/features/drupal.org/always_visible/always_visible.feature:7]: I think I own that one!

File: includes/features/drupal.org/addtocal/addtocal.feature (hunk 1 above)
Comment (sherakama) [reply includes/features/drupal.org/addtocal/addtocal.feature]: How come?

**Pull request closed without merging**